### ISSUE TITLE
[RFC] plugins should support a `--help` flag

### ISSUE THREAD

**Issue by thaJeztah** [docker/cli#1682]
relates to https://github.com/docker/cli/issues/1661 and https://github.com/docker/cli/issues/1534

One thing I just noticed;

```
Usage:	docker [OPTIONS] COMMAND

A self-sufficient runtime for containers

...

Commands:
...
  commit                    Create a new image from a container's changes
  compose     (Docker)      Define and run multi-container applications

Run 'docker COMMAND --help' for more information on a command.
```

The `help` / `usage` output states that a `COMMAND` should handle a `--help` flag, so should we define that a plugin should support that flag (at least at their top-level command) as part of the "api"?

**Comment by ijc** [comment]
At the CLI level running `docker «plugin» --help` already DTRT and calls `--help` on the plugin, it's up to the plugin what to do with it, the builtin framework/helper implements it too.

We can certainly recommend they implement `--help` (and similarly `help «command»`) but I think this is just a docs change, no code change needed.

**Comment by thaJeztah** [comment]
Yes, no code change is needed, but we should have it defined (or at least mentioned) as part of the "contract" for plugins.
